src/visitors/pages/Login.js

```diff
@@ -0,0 +1,272 @@
+/*
+ *           ---- Scan QR code ---                       ---- Load activities ---- Register visit
+ *          /                     \                    /
+ * Login --                        -- Get visitor ID --
+ *          \                     /                    \
+ *           --- Enter details ---                        ---- Failure screen
+ */
+import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { withRouter } from 'react-router-dom';
+import moment from 'moment';
+import styled from 'styled-components';
+import { Row, Col, Grid } from 'react-flexbox-grid';
+import QRPrivacy from '../components/qrprivacy';
+import QrScanner from '../components/QrScanner';
+import SignInForm from '../components/SignInForm';
+import ActivitiesGrid from '../components/ActivitiesGrid';
+import { Activities, Visitors, CbAdmin, CommunityBusiness } from '../../api';
+import { Heading, Paragraph } from '../../shared/components/text/base';
+import { PrimaryButton } from '../../shared/components/form/base';
+import NavHeader from '../../shared/components/NavHeader';
+import { redirectOnError } from '../../util';
+
+
+const StyledNav = styled.nav`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+`;
+
+const StyledSection = styled.section`
+  margin: ${props => props.margin}rem 0;
+  display: flex;
+  justify-content: center;
+`;
+
+const capitaliseFirstName = name => name.split(' ')[0].replace(/\b\w/g, l => l.toUpperCase());
+const greetingWithOrgName = orgName => `Welcome${orgName ? ` to ${orgName}` : ''}!`;
+
+export default class Login extends Component {
+  constructor() {
+    super();
+
+    this.state = {
+      initialFetchDone: false,
+      hasScanned: false,
+      visitorId: null,
+      visitorName: '',
+      orgName: '',
+      qrCodeContent: '',
+      activities: [],
+      errors: {},
+    };
+  }
+
+  componentDidMount() {
+    CbAdmin.downgradePermissions()
+      .then(() => Promise.all([
+        Activities.get({ day: 'today' }),
+        CommunityBusiness.get({ fields: ['name'] }),
+      ]))
+      .then(([resA, resC]) => {
+        this.setState({
+          initialFetchDone: true,
+          activities: resA.data.result,
+          orgName: resC.data.result.name,
+        });
+      })
+      .catch(error => redirectOnError(this.props.history.push, error));
+  }
+
+  onCameraError = () => {
+    this.props.history.push('/visitor/qrerror?e=camera');
+  }
+
+  onScannerError = () => {
+    this.props.history.push('/visitor/qrerror?e=scanner');
+  }
+
+  onSignInSuccess = (res) => {
```

Comment:
Not explicit what's happening here, at least destructure arguments for more clarity.